src/Persisters/Flysystem.php

```diff
@@ -162,17 +126,23 @@ public function save(Persistable $persistable) : void
      */
     public function load() : Persistable
     {
-        if (!$this->filesystem->has($this->path)) {
-            throw new RuntimeException("File does not exist in filesystem. {$this}");
+        if (!$this->filesystem->fileExists($this->path)) {
+            throw new RuntimeException("File does not exist at {$this->path}.");
+        }
+
+        try {
+            $data = $this->filesystem->read($this->path);
+        } catch (FilesystemException $e) {
+            throw new RuntimeException("Error reading data from {$this->path}.");
         }
```

Comment:
As above, let's start the `try` block at start of this method so that anything raised from `fileExists` is also handled.